core/src/main/java/org/mule/runtime/core/internal/policy/PolicyNextActionMessageProcessor.java

```diff
@@ -117,7 +117,7 @@ public Publisher<CoreEvent> apply(Publisher<CoreEvent> eventPub) {
                 : policyEventMapper.onOperationPolicyNext(event))
             .transform((ReactiveProcessor) ((Reference) ctx.get(POLICY_NEXT_OPERATION)).get()));
       }
-    }), policyNextErrorHandler(), NO_EXPORT_CHILD_NAMED_SPAN_BASED_ON_PARENT_SPAN_CHILD_SPAN_CUSTOMIZATION_INFO);
+    }), policyNextErrorHandler(), initialSpanInfoBuilderProvider.getComponentInitialSpanInfoBuilder(this).withNoExport().build());
```

Comment:
This is where I think that a break of encapsulation is seen. If we are going to base the export levels feature on the existing InitialExportInfo. a withNoExport method should not be  present on the builders of the InitialSpanInfo but be part of the different export strategies logic. If this is meant to be changed in next PRs lets add a TODO with the task ID and move on, it's ok to me, but I think that it would be better to have an getInitialExportInfoProvider than a getInitialSpanInfoBuilerProvider, because we are not changing anything from the InitialSpanInfo but it's composed InitialExportInfo.

Reply:
I've changed this. In the "execution engine" you directly retrieve the InitialSpanInfo from an injectable (managed) component without indicating anything regarding the exportability.

Within the implementation for the InitialSpanInfoProvider you have a component ExportInitialSpanInfoProvider that is responsibility for indicating if it is exportable and how.

For the moment an implementation for the default level is implemented. As for the roadmap the other implementations will be handled in [User Story B](https://github.com/mulesoft/mule/pull/12316/files#diff-0ea6de91b7f846fa6f2ce6c659bdd5f2c8a406f06144f1e83a2783a91bb5540cR24) (next sprint).

And eventually this is component will be inyected according to the retrievable configuration (verify the user stories for that)